types/timestamp_test.go

```diff
@@ -96,3 +98,27 @@ func TestProtobufToGogoTimestamp(t *testing.T) {
 		})
 	}
 }
+
+func TestParseDate(t *testing.T) {
+	tcs := []struct {
+		name   string
+		date   string
+		hasErr bool
+	}{
+		{"good", "2022-01-20", false},
+		{"bad", "01-2021-20", true},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			require := require.New(t)
+			tm, err := types.ParseDate(tc.date, tc.date)
+			if tc.hasErr {
+				require.Error(err)
+				require.Nil(tm)
```

Comment:
We have to use the zero value of `time.Time`, which is also returned when `time.Parse` returns an error.

```suggestion
				require.Equal(time.Time{}, tm)
```